fogros2/fogros2/aws.py

```diff
@@ -252,16 +235,31 @@ def __init__(self, ami_image, region="us-west-1", ec2_instance_type="t2.micro",
         self.ec2_instance_disk_size = disk_size  # GB
         self.aws_ami_image = ami_image
 
-        # key & security group names
-        self.ec2_security_group = "FOGROS2_SECURITY_GROUP"
-        self.ec2_key_name = "FogROS2KEY-" + self.unique_name
-        self.ssh_key_path = os.path.join(self.working_dir, self.ec2_key_name + ".pem")
-
         # aws objects
         self.ec2_instance = None
         self.ec2_resource_manager = boto3.resource("ec2", self.region)
         self.ec2_boto3_client = boto3.client("ec2", self.region)
 
+        # Check for name collision among other AWS instances
+        cur_instances = self.ec2_boto3_client.describe_instances(
+            Filters=[
+                {"Name": "instance.group-name", "Values": ["FOGROS2_SECURITY_GROUP"]}
+            ]
+        )
+        cur_instance_names = []
+        for res in cur_instances["Reservations"]:
+            for inst in res["Instances"]:
+                tag_map = {t["Key"]: t["Value"] for t in inst["Tags"]} if "Tags" in inst else {}
+                if "FogROS2-Name" in tag_map:
+                    cur_instance_names.append(tag_map["FogROS2-Name"])
+        while self.name in cur_instance_names:
```

Comment:
This is a race-condition, but extremely unlikely to occur.  If another fogros concurrently starts with the same name between line 244 and here, it will result in an undetected name collision here.

Reply:
Agree; I felt the tradeoff for speed by not querying every time was worth it. Plus we don't actually get knowledge of the new instance until it starts, so the condition you're mentioning is still possible even by querying every time if someone starts two instances at roughly the same time (both would query and not find the collision).

Reply:
This is resolved in the updated version